bson-record-codec/src/main/org/bson/codecs/record/annotations/BsonId.java

```diff
@@ -26,7 +26,9 @@
  * An annotation that configures the record component as the _id field of the document
  *
  * @since 4.6
+ * @deprecated Prefer {@link org.bson.codecs.pojo.annotations.BsonId}
```

Comment:
I propose updating the documentation for all annotations in `org.bson.codecs.pojo.annotations` to reflect the following:

- Whether the annotation may be used with a record.
- How to use it with a record (the only valid place for currently supported annotations is a record component).

Without such docs, runtime exceptions is the only way for users to learn.

Reply:
Done